backend/port_monitor.py

```
@@ -352,116 +357,156 @@ def restart_container(self, container_name: str) -> bool:
+                    port_ok = True
+                    break
+                await asyncio.sleep(5)
+
+            if not port_ok:
+                client = self.get_docker_client()
+                running = False
+                if client:
+                    try:
+                        container = client.containers.get(stack.primary_container)
+                        running = container.status == "running"
+                    except Exception:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)